### DISCUSSION TITLE
strings containing qoute cause #MATH behave unexpectedly

### DISCUSSION THREAD

**Discussion by bbcallen** [scandum/tintin#84] [Bugs]
ver 2.02.12 from BREW install on macos Big Sur 11.6.1
Not sure if it can be reproduced on other platforms.

===== compare string containg quote =====
#VA a {do something ""}
#OK. VARIABLE {a} HAS BEEN SET TO {do something ""}.
#MATH m {"$a" != ""}
#MATH: VARIABLE {m} HAS BEEN SET TO {0}.
#MATH m {"$a" == ""}
#MATH: VARIABLE {m} HAS BEEN SET TO {0}.

Both != and == result 0.

===== normal compare =====
#VA a {do something}
#OK. VARIABLE {a} HAS BEEN SET TO {do something}.
#MATH m {"$a" != ""}
#MATH: VARIABLE {m} HAS BEEN SET TO {1}.
#MATH m {"$a" == ""}
#MATH: VARIABLE {m} HAS BEEN SET TO {0}.

!== and === do same as != and ==

This problem also affect #IF.

**Comment by scandum** [comment]
Best workaround when there can/will be quotes in the string comparison is to use braces instead of quotes.
```
#VA a {do something ""}
#OK. VARIABLE {a} HAS BEEN SET TO {do something ""}.
#MATH m {"$a" != ""}
#MATH: VARIABLE {m} HAS BEEN SET TO {0}.
#MATH m {{$a} != {}}
#MATH: VARIABLE {m} HAS BEEN SET TO {1}.
```
Another option is to encode the quotes as hex.
```
#VA a {do something \x22\x22}
#OK. VARIABLE {a} HAS BEEN SET TO {do something \x22\x22}.
#MATH m {"$a" != ""}
#MATH: VARIABLE {m} HAS BEEN SET TO {1}.
```

**Comment by bbcallen** [comment]
Braces work for me.

Thanks for your reply.
